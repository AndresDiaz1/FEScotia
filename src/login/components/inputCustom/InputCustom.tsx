import React, { useState, useEffect } from 'react';
import { IonItem, IonLabel, IonInput } from '@ionic/react';
import { IconProp } from '@fortawesome/fontawesome-svg-core';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { InputChangeEventDetail, TextFieldTypes } from './InputCustomTypes';

interface InputFloatingLabelProps {
  initialValue?: string;
  label: string;
  onChange: (event: CustomEvent<InputChangeEventDetail>) => void;
  type?: TextFieldTypes;
  containerClassName?: string;
  inputTypeHandler?: () => void;
  iconName?: IconProp;
}

const InputFloatingLabel: React.FC<InputFloatingLabelProps> = ({
  initialValue = '',
  label,
  type = 'text',
  containerClassName = '',
  onChange,
  inputTypeHandler,
  iconName,
}: InputFloatingLabelProps) => {
  const [inputValue, setInputValue] = useState(initialValue);

  useEffect(() => {
    setInputValue(initialValue);
  }, [initialValue]);

  return (
    <IonItem className={containerClassName}>
      {inputTypeHandler ? (
        <div onClick={inputTypeHandler} className="login-page__reveal-password-container">
          <FontAwesomeIcon className="login-page__reveal-password-icon" icon={iconName as IconProp} />
        </div>
      ) : (
        ''
      )}
      <IonLabel className={inputValue ? '' : 'ion-hide'} position="floating">
        {label}
      </IonLabel>
      <IonInput
        value={inputValue}
        type={type}
        autofocus={true}
        onIonChange={(event) => {
          const {
            detail: { value },
          } = event;
          if (value != null) setInputValue(value);
          onChange(event);
        }}
        placeholder={label}
        name={label}
      />
    </IonItem>
  );
};

export default InputFloatingLabel;
