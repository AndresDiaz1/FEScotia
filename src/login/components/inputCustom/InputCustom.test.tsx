import React from 'react';
import { mount } from 'enzyme';
import InputCustom from './InputCustom';

describe('Input custom Component', () => {
    describe('Initial state', ()=> {
        it('Should render', ()=> {
            const component = mount(<InputCustom label={'Test'} onChange={()=>console.log('Test change')}/>);
            expect(component).toMatchSnapshot();
        });
    });
});