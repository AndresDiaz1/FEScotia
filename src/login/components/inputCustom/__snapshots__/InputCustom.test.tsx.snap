// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input custom Component Initial state Should render 1`] = `
<InputFloatingLabel
  label="Test"
  onChange={[Function]}
>
  <ForwardRef(IonItem)
    className=""
  >
    <IonItem
      className=""
      forwardedRef={null}
    >
      <ion-item
        onClick={[Function]}
      >
        <ForwardRef(IonLabel)
          className="ion-hide"
          position="floating"
        >
          <IonLabel
            className="ion-hide"
            forwardedRef={null}
            position="floating"
          >
            <ion-label
              position="floating"
            >
              Test
            </ion-label>
          </IonLabel>
        </ForwardRef(IonLabel)>
        <ForwardRef(IonInput)
          autofocus={true}
          name="Test"
          onIonChange={[Function]}
          placeholder="Test"
          type="text"
          value=""
        >
          <IonInput
            autofocus={true}
            forwardedRef={null}
            name="Test"
            onIonChange={[Function]}
            placeholder="Test"
            type="text"
            value=""
          >
            <ion-input
              name="Test"
              placeholder="Test"
              type="text"
              value=""
            />
          </IonInput>
        </ForwardRef(IonInput)>
      </ion-item>
    </IonItem>
  </ForwardRef(IonItem)>
</InputFloatingLabel>
`;
